python/cuml/test/test_naive_bayes.py

```diff
@@ -234,3 +237,80 @@ def test_score(x_dtype, y_dtype, nlp_20news):
     THRES = 1e-4
 
     assert sk_score - THRES <= cuml_score <= sk_score + THRES
+
+
+@pytest.mark.parametrize("x_dtype", [cp.float32, cp.float64])
+@pytest.mark.parametrize("y_dtype", [cp.int32, cp.int64])
+@pytest.mark.parametrize("is_sparse", [True, False])
+def test_bernoulli(x_dtype, y_dtype, is_sparse, nlp_20news):
+    X, y = nlp_20news
+    n_rows = 500
+
+    X = sparse_scipy_to_cp(X, x_dtype).astype(x_dtype)
+    y = y.astype(y_dtype)
+
+    X = X.tocsr()[:n_rows]
+    y = y[:n_rows]
+    if not is_sparse:
+        X = X.todense()
+
+    sk_model = skBNB()
+    cuml_model = BernoulliNB()
+
+    sk_model.fit(X.get(), y.get())
+    cuml_model.fit(X, y)
+
+    sk_score = sk_model.score(X.get(), y.get())
+    cuml_score = cuml_model.score(X, y)
+    cuml_proba = cuml_model.predict_log_proba(X).get()
+    sk_proba = sk_model.predict_log_proba(X.get())
+
+    THRES = 1e-3
+
+    assert_array_equal(sk_model.class_count_, cuml_model.class_count_.get())
+    assert_allclose(sk_model.class_log_prior_,
+                    cuml_model.class_log_prior_.get(), 1e-6)
+    assert_allclose(cuml_proba, sk_proba, atol=1e-2, rtol=1e-2)
+    assert sk_score - THRES <= cuml_score <= sk_score + THRES
+
+
+@pytest.mark.parametrize("x_dtype", [cp.float32, cp.float64])
+@pytest.mark.parametrize("y_dtype", [cp.int32,
+                                     cp.float32, cp.float64])
+def test_bernoulli_partial_fit(x_dtype, y_dtype, nlp_20news):
+    chunk_size = 500
+
+    X, y = nlp_20news
+
+    X = sparse_scipy_to_cp(X, x_dtype).astype(x_dtype)
+    y = y.astype(y_dtype)
+
+    X = X.tocsr()
+
+    model = BernoulliNB()
+
+    classes = np.unique(y)
+
+    for i in range(math.ceil(X.shape[0] / chunk_size)):
+
+        upper = i*chunk_size+chunk_size
+        if upper > X.shape[0]:
+            upper = -1
+
+        if upper > 0:
+            x = X[i*chunk_size:upper]
+            y_c = y[i*chunk_size:upper]
+        else:
+            x = X[i*chunk_size:]
+            y_c = y[i*chunk_size:]
+
+        model.partial_fit(x, y_c, classes=classes)
+        if upper == -1:
+            break
+
+    y_hat = model.predict(X)
+
+    y_hat = cp.asnumpy(y_hat)
+    y = cp.asnumpy(y)
+
+    assert accuracy_score(y, y_hat) >= 0.78
```

Comment:
Same here, a better comparison would be between cpu and gpu, that way we know they match.